var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -575,12 +623,25 @@ def edit(self, spec, prefix):
                     cppflags += ["-D__PW_CUDA"]
                     libs += ["-lcufft", "-lcublas"]
 
-            cuda_arch = spec.variants["cuda_arch"].value[0]
-            if cuda_arch:
-                gpuver = {"35": "K40", "37": "K80", "60": "P100", "70": "V100"}[cuda_arch]
+            gpuver = gpu_map[spec.variants["cuda_arch"].value]
+            if cuda_arch == "35" and spec.satisfies("+cuda_arch_35_k20x"):
+                gpuver = "K20X"
+
+        if spec.satisfies(+rocm) and spec.satisfies("@2022:"):
+            libs += [
+                "-L{}".format(spec["rocm"].libs.directories[0]),
+                "-L{}/stubs".format(spec["rocm"].libs.directories[0]),
+                "-lhipblas",
+                "-lhipfft",
+                "-lstdc++",
+            ]
 
-                if cuda_arch == "35" and spec.satisfies("+cuda_arch_35_k20x"):
-                    gpuver = "K20X"
+            cppflags += ["-D__OFFLOAD_HIP"]
+            acc_compiler_var = "hipcc"
+            acc_flags_var = "NVFLAGS"
+            cppflags += ["-D__ACC"]
+            cppflags += ["-D__DBCSR_ACC"]
+            gpuver = gpu_map[spec.variants["amdgpu_target"].value]
```

Comment:
```suggestion
            gpuver = gpu_map[spec.variants["amdgpu_target"].value[0]]
```

Reply:
Also, does `gpuver` need to be used somewhere in the `+rocm` case? I can't see it being used but I may have missed it...